plugins/chanbackup.c

```diff
@@ -320,6 +320,74 @@ static void update_scb(struct plugin *p, struct scb_chan **channels)
 	rename("scb.tmp", "emergency.recover");
 }
 
+
+static struct command_result
+*after_send_their_peer_strg(struct command *cmd,
+                            const char *buf,
+                            const jsmntok_t *params,
+                            void *cb_arg UNUSED)
+{
+        plugin_log(cmd->plugin, LOG_DBG, "Sent their peer storage!");
+	return notification_handled(cmd);
+}
+
+static struct command_result *after_listdatastore(struct command *cmd,
+                                                  const char *buf,
+                                                  const jsmntok_t *params,
+                                                  struct node_id *nodeid)
+{
+        const jsmntok_t *datastoretok = json_get_member(buf,
+                                                        params,
+                                                        "datastore");
+
+        if (datastoretok->size == 0)
+        	return notification_handled(cmd);
+
+        const jsmntok_t *payloadtok = json_get_member(buf,
+                                                      json_get_arr(datastoretok,
+                                                                   0),
+                                                      "hex");
+        struct out_req *req;
+
+        u8 *hexdata = tal_hexdata(cmd,
+                                  json_strdup(cmd, buf, payloadtok),
+                                  payloadtok->end - payloadtok->start),
+        *payload = towire_your_peer_storage(cmd, hexdata);
+
+        plugin_log(cmd->plugin, LOG_DBG,
+                   "sending their backup from our datastore");
+
+        req = jsonrpc_request_start(cmd->plugin,
+                                    cmd,
+                                    "sendcustommsg",
+                                    after_send_their_peer_strg,
+                                    &forward_error,
+                                    NULL);
+
+        json_add_node_id(req->js, "node_id", nodeid);
+        json_add_hex(req->js, "msg", payload,
+                     tal_bytelen(payload));
+
+        return send_outreq(cmd->plugin, req);
+}
+
+static struct command_result *after_send_scb(struct command *cmd,
+                                             const char *buf,
+                                             const jsmntok_t *params,
+                                             struct node_id *nodeid)
+{
+        plugin_log(cmd->plugin, LOG_DBG, "Peer storage sent!");
+        struct out_req *req;
+        req = jsonrpc_request_start(cmd->plugin,
+                                    cmd,
+                                    "listdatastore",
+                                    after_listdatastore,
+                                    &forward_error,
+                                    nodeid);
+        json_add_node_id(req->js,"key",nodeid);
```

Comment:
We want a path here, not just top-level keys in the datrastore!

i.e. ["chanbackup", "peers", nodeid].  In fact, you can see how autoclean does it with "datastore_path", though we do't have datastore helpers for this.  Let me write some...